corehq/apps/hqmedia/models.py

```diff
@@ -942,29 +942,40 @@ def get_media_objects(self, build_profile_id=None):
         found_missing_mm = False
         # preload all the docs to avoid excessive couch queries.
         # these will all be needed in memory anyway so this is ok.
-        expected_ids = [map_item.multimedia_id for map_item in self.multimedia_map.values()]
-        raw_docs = dict((d["_id"], d) for d in iter_docs(CommCareMultimedia.get_db(), expected_ids))
-        all_media = {m.path: m for m in self.all_media()}
-        deleted_media = []
-        allow_deletion = not toggles.CAUTIOUS_MULTIMEDIA.enabled(self.domain)
         build_profile = self.build_profiles[build_profile_id] if build_profile_id else None
-        for path, map_item in self.multimedia_map_for_build(build_profile=build_profile).items():
+        multimedia_map_for_build = self.multimedia_map_for_build(build_profile=build_profile)
+        expected_ids = [map_item.multimedia_id for map_item in multimedia_map_for_build.values()]
+        raw_docs = dict((d["_id"], d) for d in iter_docs(CommCareMultimedia.get_db(), expected_ids))
+        retry_successes = []
+        retry_failures = []
+        for path, map_item in multimedia_map_for_build.items():
             media_item = raw_docs.get(map_item.multimedia_id)
             if media_item:
                 media_cls = CommCareMultimedia.get_doc_class(map_item.media_type)
                 yield path, media_cls.wrap(media_item)
             else:
-                # delete media reference from multimedia map so this doesn't pop up again!
-                deleted_media.append((path, map_item, all_media.get(path)))
-                if allow_deletion:
-                    del self.multimedia_map[path]
-                    found_missing_mm = True
-
-        if not allow_deletion and deleted_media:
-            _log_media_deletion(self, deleted_media)
-
-        if found_missing_mm:
-            self.save()
+                # Re-attempt to fetch media directly from couch
+                media = CommCareMultimedia.get_doc_class(map_item.media_type)
+                try:
+                    media = media.get(map_item.multimedia_id)
+                    retry_successes.append(map_item)
+                    yield path, media
+                except ResourceNotFound:
```

Comment:
Should we just raise an error instead here after the retry? Don't think its worth ignoring and let the user continue with doing what they are doing

Reply:
a90973e40fa - I'm wary of changing the default behavior, but happy to make it fail hard just for ICDS.

Reply:
I would still push(not strongly) for making this generic. For some reason this has been an issue with only ICDS but its a possibility for any domain.

Reply:
I'm going to hold off for now, will keep this email in my inbox to reconsider in a couple of weeks (maybe at that point we'll have a better handle on what's causing all of this...).